src/app/3d/qgsmaptoolclippingplanes.cpp

```diff
@@ -0,0 +1,188 @@
+  mRubberBandPoints.reset( new QgsRubberBand( canvas, Qgis::GeometryType::Point ) );
+  mRubberBandPoints->setColor( QColorConstants::Red );
+  mRubberBandPoints->setIconSize( 10 );
+  mRubberBandLines->setColor( QColorConstants::Red );
+  mRubberBandLines->setWidth( 3 );
+  QColor polygonColor = QColorConstants::Red.lighter();
+  polygonColor.setAlphaF( 0.5 );
+  mRubberBandPolygon->setColor( polygonColor );
+}
+
+void QgsMapToolClippingPlanes::activate()
+{
+  QgsMapTool::activate();
+  mRubberBandPoints->show();
+  mRubberBandLines->show();
+  mRubberBandPolygon->show();
+}
+
+void QgsMapToolClippingPlanes::keyReleaseEvent( QKeyEvent *e )
+{
+  if ( e->key() == Qt::Key_Escape )
+  {
+    clear();
+  }
+}
+
+void QgsMapToolClippingPlanes::deactivate()
+{
+  clearRubberBand();
+  QgsMapTool::deactivate();
+}
+
+void QgsMapToolClippingPlanes::canvasMoveEvent( QgsMapMouseEvent *e )
+{
+  if ( mClicked || mRubberBandPoints->numberOfVertices() == 0 )
+    return;
+
+  const QgsPointXY point = toMapCoordinates( e->pos() );
+  if ( mRubberBandPoints->numberOfVertices() == 2 )
+  {
+    QgsVector vec( *mRubberBandPoints->getPoint( 0, 1 ) - *mRubberBandPoints->getPoint( 0, 0 ) );
+    vec = vec.normalized().perpVector();
+    mRectangleWidth = QgsGeometryUtils::distToInfiniteLine(
+      QgsPoint( point ),
+      QgsPoint( *mRubberBandPoints->getPoint( 0, 0 ) ),
+      QgsPoint( *mRubberBandPoints->getPoint( 0, 1 ) )
+    );
+    const QVector<QgsPointXY> points( {
+      *mRubberBandPoints->getPoint( 0, 0 ) + vec * mRectangleWidth, //! top left corner
+      *mRubberBandPoints->getPoint( 0, 1 ) + vec * mRectangleWidth, //! top right corner
+      *mRubberBandPoints->getPoint( 0, 1 ) - vec * mRectangleWidth, //! bottom right corner
+      *mRubberBandPoints->getPoint( 0, 0 ) - vec * mRectangleWidth  //! bottom left corner
+    } );
+    mRubberBandPolygon->setToGeometry( QgsGeometry( new QgsPolygon( new QgsLineString( points ) ) ) );
+  }
+  else
+  {
+    mRubberBandLines->movePoint( point );
+  }
+}
+
+void QgsMapToolClippingPlanes::canvasPressEvent( QgsMapMouseEvent * )
+{
+  if ( mRubberBandPoints->numberOfVertices() == 0 )
+  {
+    clearHighLightedArea();
+  }
+  mClicked = true;
+}
+
+void QgsMapToolClippingPlanes::canvasReleaseEvent( QgsMapMouseEvent *e )
+{
+  if ( e->button() == Qt::LeftButton )
+  {
+    const QgsPointXY point = toMapCoordinates( e->pos() );
+    if ( mRubberBandPoints->numberOfVertices() == 2 )
+    {
+      //check if cross-section is in canvas extents
+      const QgsGeometry crossSectionPolygon = mRubberBandPolygon->asGeometry();
+      if ( !crossSectionPolygon.intersects( m3DCanvas->mapCanvas3D()->scene()->sceneExtent() ) )
+      {
+        clear();
+        QgsMessageBar *msgBar = QgisApp::instance()->messageBar();
+        msgBar->pushInfo( QString(), tr( "The cross section is outside of the scene extent, please select a new one!" ) );
```

Comment:
nitpick. No need for a `of` here: `The cross section is outside the scene extent, please select a new one!`